doc/source/ray-job-submission/overview.rst

```diff
@@ -0,0 +1,243 @@
+
+
+    ❯ ray job logs raysubmit_GsQYzyvZpgNicU8F
+    123
+
+
+Ray Job SDK
+------------
+
+Ray Job SDK is the recommended way to submit Jobs programmatically.
+
+We can import and intialize the Job submission client by providing an valid Ray cluster head node address where the port is same as the port used by Ray dashboard. We're using your local Ray cluster as an example but it works the same for remote Ray cluster addresses.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.sdk import JobSubmissionClient
+
+    client = JobSubmissionClient("http://127.0.0.1:8265")
+
+Then we can submit our application to the Ray cluster via the Job SDK.
+
+.. code-block:: python
+
+    job_id = client.submit_job(
+        # Entrypoint shell command to execute
+        entrypoint="python script.py",
+        # Working dir
+        runtime_env={
+            "working_dir": "./",
+            "pip": ["requests==2.26.0"]
+        }
+    )
+
+.. tip::
+
+    By default Ray Job server will generate a new uuid as return value, but you can also generate your unique job_id first and pass it into :code:`submit_job`. In this case the Job will be executed with your given id, and will throw error if same job_id is submitted more than once for the same Ray cluster. This can facilitate Job id management assuming you have your own Job registry to generate and persist unique Job ids.
+
+Now we can have a simple polling loop that checks the job status until it reaches a terminal state (namely, ``JobStatus.SUCCEEDED``, ``JobStatus.STOPPED``, or ``JobStatus.FAILED``), and gets the logs at the end. We expect to see actor printed numbers as well as the correct version of the :code:`requests` module specified in the ``runtime_env``.
+
+.. code-block:: python
+
+    from ray.dashboard.modules.job.common import JobStatus, JobStatusInfo
+
+    def wait_until_finish(job_id):
+        start = time.time()
+        timeout = 5
+        while time.time() - start <= timeout:
+            status_info = client.get_job_status(job_id)
+            status = status_info.status
+            print(f"status: {status}")
+            if status in {JobStatus.SUCCEEDED, JobStatus.STOPPED, JobStatus.FAILED}:
+                break
+            time.sleep(1)
+
+
+    wait_until_finish(job_id)
+    logs = client.get_job_logs(job_id)
+    print(logs)
+
+.. tip::
+
+    We can also use other remote uris for runtime env, such as S3 or Git. See "Remote URIs" section of :ref:`Runtime Environments<runtime-environments>` for details.
+
+A submitted Job can be stopped by the user before it finishes executing.
+
+.. code-block:: python
+
+    job_id = client.submit_job(
+        # Entrypoint shell command to execute
+        entrypoint="python -c 'import time; time.sleep(60)'",
+        runtime_env={}
+    )
+    wait_until_finish(job_id)
+    client.stop_job(job_id)
+    wait_until_finish(job_id)
+    logs = client.get_job_logs(job_id)
+    print(logs)
```

Comment:
add some prints so this shows something?

Reply:
i ended up removing it from the snippet, in the future i think we can have a separate e2e notebook with all intermediate outputs like https://gist.github.com/jiaodong/17a2c34876fdea6d2eabb2c8ca04b3ff